Marlin/Configuration_adv.h

```diff
@@ -2275,6 +2275,135 @@
   #define E6_HYBRID_THRESHOLD     30
   #define E7_HYBRID_THRESHOLD     30
 
+  /**
+   * CoolStep. TMC2209 only.
+   * This mode allows for cooler steppers and energy savings.
+   * STEALTHCHOP_(XY|Z|E) must be enabled to use CoolStep.
+   * The driver will switch to coolStep when stepper speed is over COOLSTEP_THRESHOLD mm/s.
+   * If SG_RESULT goes below COOLSTEP_LOWER_LOAD_THRESHOLD stepper curreent will be increased.
+   * If SG_RESULT goes above COOLSTEP_UPPER_LOAD_THRESHOLD stepper curreent will be decreased.
+   * SEUP sets the increase step width. Value range is 0..3 and computed as 2^SEUP.
+   * SEDN sets the decrease delay. Value range is 0..3, 0 being the slowest.
+   * SEIMIN sets the lower current limit. 0: 1/2 of IRUN, 1:1/4 of IRUN   
+   */
+  
+  #if AXIS_IS_TMC(X)
```

Comment:
If you look in `core/drivers.h` there are now macros there such as `AXIS_HAS_STALLGUARD` and `AXIS_HAS_STEALTHCHOP`. You could easily add an `AXIS_HAS_COOLSTEP` macro, which would further enforce the 2209 restriction here.

Reply:
Last I checked the 2130 also has coolstep. But does it also support setting these parameters?

Reply:
Yes 2130 does have coolstep and the settings are the same, the main difference is that in the 2130 coolstep can only be used in conjunction with spreadcycle which is less useful to us mainly operating on the stealthchop range. If you wish I can implement it for the 2130 as well though I cant test it as I don't have that driver.